pyvista/themes.py

```diff
@@ -0,0 +1,2202 @@
+        or ``'paraview'``.
+
+    Examples
+    --------
+    Set to the default theme.
+
+    >>> import pyvista
+    >>> pyvista.set_plot_theme('default')
+
+    Set to the document theme.
+
+    >>> pyvista.set_plot_theme('document')
+
+    Set to the dark theme.
+
+    >>> pyvista.set_plot_theme('dark')
+
+    Set to the ParaView theme.
+
+    >>> pyvista.set_plot_theme('paraview')
+
+    """
+    import pyvista
+    if isinstance(theme, str):
+        theme = theme.lower()
+        if theme == 'night':  # pragma: no cover
+            warnings.warn('use "dark" instead of "night" theme', PyvistaDeprecationWarning)
+            theme = 'dark'
+        new_theme_type = _ALLOWED_THEMES[theme].value
+        pyvista.global_theme.load_theme(new_theme_type())
+    elif isinstance(theme, DefaultTheme):
+        pyvista.global_theme.load_theme(theme)
+    else:
+        raise TypeError(f'Expected a ``pyvista.DefaultTheme`` or ``str``, not '
+                        f'a {type(theme).__name__}')
+
+
+class _ThemeConfig():
+    """Provide common methods for theme configuration classes."""
+
+    __slots__: List[str] = []
+
+    @classmethod
+    def from_dict(cls, dict_):
+        """Create from a dictionary."""
+        inst = cls()
+        for key, value in dict_.items():
+            if not hasattr(inst, key):
+                raise KeyError(f'Invalid key "{key}" for {cls.__name__}')
+            attr = getattr(inst, key)
+            if hasattr(attr, 'from_dict'):
+                setattr(inst, key, attr.from_dict(value))
+            else:
+                setattr(inst, key, value)
+        return inst
+
+    def to_dict(self) -> dict:
+        """Return theme config parameters as a dictionary."""
+        # remove the first underscore in each entry
+        dict_ = {}
+        for key in self.__slots__:
+            value = getattr(self, key)
+            key = key[1:]
+            if hasattr(value, 'to_dict'):
+                dict_[key] = value.to_dict()
+            else:
+                dict_[key] = value
+        return dict_
+
+    def __eq__(self, other):
+        if not isinstance(self, type(other)):
+            return False
+
+        for attr_name in other.__slots__:
+            attr = getattr(self, attr_name)
+            other_attr = getattr(other, attr_name)
+            if isinstance(attr, (tuple, list)):
+                if tuple(attr) != tuple(other_attr):
+                    return False
+            else:
+                if not attr == other_attr:
+                    return False
+
+        return True
+
+    def __getitem__(self, key):
+        """Get a value via a key.
+
+        Implemented here for backwards compatibility.
+        """
+        return getattr(self, key)
+
+    def __setitem__(self, key, value):
+        """Set a value via a key.
+
+        Implemented here for backwards compatibility.
+        """
+        setattr(self, key, value)
```

Comment:
Very smart!

Reply:
I wanted to maintain backwards compatibility.  Not sure how many are using themes, but some of my other modules use it and it's always a pain to change.

Reply:
Also, I changed the warnings back to `DepricationWarning` so we don't immediately start warning users of the change.  If we wait a few releases, most should be using `pyvista.global_theme` anyway.

Reply:
OK, but I added some warning checks to the backwards compat theme test that have to be changed. If we now have proper deprecation warnings we can use `with pytest.deprecated_call()`.

Reply:
But I now see you've removed those from the test. That's fine, but this is also a reminder that the deprecation warnings are not visible by default at all.